americangut/parallel.py

```diff
@@ -0,0 +1,31 @@
+import multiprocessing as mp
+
+import americangut.results_utils as agru
+
+
+def dispatcher(success_fp, fail_fp, partitions):
+    """Dispatch execution over a pool of processors
+
+    Parameters
+    ----------
+    success_fp : file-like object
+        A file-like object to write a list of successful sample IDs too
+    fail_fp : file-like object
+        A file-like object to write a list of unsuccessful sample IDs too,
+        and any associated error information
+    partitions : iterable
+        Yields a function and an iterable of IDs. It is expected that this
+        function will return a dict of {str: list}.
+    """
+    pool = mp.Pool(processes=mp.cpu_count())
+
```

Comment:
It introduces either a config file or yet another env var.
On Oct 19, 2015 12:29 PM, "mortonjt" <notifications@github.com> wrote:

> In americangut/parallel.py
> <https://github.com/biocore/American-Gut/pull/160#discussion_r42408002>:
>
> > +def dispatcher(success_fp, fail_fp, partitions):
> > +    """Dispatch execution over a pool of processors
> > +
> > +    Parameters
> > +    ----------
> > +    success_fp : file-like object
> > +        A file-like object to write a list of successful sample IDs too
> > +    fail_fp : file-like object
> > +        A file-like object to write a list of unsuccessful sample IDs too,
> > +        and any associated error information
> > +    partitions : iterable
> > +        Yields a function and an iterable of IDs. It is expected that this
> > +        function will return a dict of {str: list}.
> > +    """
> > +    pool = mp.Pool(processes=mp.cpu_count())
> > +
>
> Would it be worthwhile to pass in the number of cores as a parameter?
>
> That why the user can specify exactly how many cores that they would like
> to use. This could also help prevent users from accidentally over-swamping
> their system.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/American-Gut/pull/160/files#r42408002>.
>